enhancements/ingress/optional-ingress-hypershift.md

```diff
@@ -0,0 +1,319 @@
+- AWS load balancer controller provisions an ALB on AWS.
+- Cluster instance admin can install [External DNS Operator](/enhancements/dns/externaldns-operator.md) to help cluster instance users set up convenient DNS names.
+
+### API Extensions
+
+N/A - no api extensions are being introduced in this proposal
+
+### Implementation Details/Notes/Constraints
+
+#### Ingress capability
+
+The name of the new capability is `Ingress`. It includes the cluster ingress operator and [all its APIs](#ingress-operator-apis) but does NOT include [the related APIs](#related-apis).
+The implementation of the new capability should follow the instructions of [how to implement a new capability](/enhancements/installer/component-selection.md#how-to-implement-a-new-capability), to name some:
+- `Ingress` capability should be added to `openshift/api` repository.
+- The ingress capability should be part of the `vCurrent` and `v4.14` baseline sets.
+- The openshift api vendored dependency should be bumped into `cluster-version-operator` repository.
+  - Additionally some more changes will need to be done in the cluster version operator to address [the standalone use case](#standalone-openshift).
+- The openshift api vendored dependency should be bumped in `installer` repository.
+  - This will allow the users of the standalone OpenShift to see the ingress capability. While the enabling of the ingress capability is not supposed to cause any issue, the disabling may have some consequences therefore will be not possible, see [Standalone OpenShift chapter](#standalone-openshift) for more details.
+
+Similar to any other capability, the ingress capability:
+- can be enabled not only as part of the capability sets (`vCurrent`, `v4,14`) but explicitly by using [the additional enabled capability field](https://github.com/openshift/api/blob/master/config/v1/types_cluster_version.go#L433-L438).
+- if enabled, should appear in [the list of the enabled capabilities](https://github.com/openshift/api/blob/master/config/v1/types_cluster_version.go#L445-L448) of `ClusterVersion` object.
+- if enabled, cannot be disabled post install.
+
+#### HyperShift
+
+HyperShift doesn't use the OpenShift installer to spawn the user clusters. HyperShift deploys the cluster version operator to manage the cluster operators though.
+However not all the cluster operators are of use for HyperShift.
+The functionality of some cluster operators is fully replaced by the logic in the hosted control plane (e.g. authentication operator).
+The ingress operator is an example of a cluster operator whose life cycle is shared between the cluster version operator and HyperShift.
+HyperShift doesn't pass all the ingress operator's payload to the cluster version operator.
+Certain manifests are filtered out, most importantly: [the ingress operator's deployment](https://github.com/openshift/hypershift/blob/main/control-plane-operator/controllers/hostedcontrolplane/cvo/reconcile.go#L75).
```

Comment:
nit: floating this link with `/main/` leaves the `L75` reference likely to go stale.  Probably worth pinning to a specific commit like https://github.com/openshift/hypershift/blob/0a5ec529ce7ae6f238ed27be6eb5757cf70c41f9/control-plane-operator/controllers/hostedcontrolplane/cvo/reconcile.go#L75 , so folks could see you were referring to `manifestsToOmit`'s `0000_50_cluster-ingress-operator_02-deployment-ibm-cloud-managed.yaml`, and then if  necessary hunt through HyperShift's subsequent Git history to figure out if/how that functionality is handled in the future.

Reply:
[Done](https://github.com/openshift/enhancements/commit/af34c77bc9dc3039bc58114573ccf4c90ddbf8ac), for all the links which use `#L` references.